src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -252,6 +252,22 @@ void IExtensionConfigProvider.Initialize(ExtensionConfigContext context)
             rule.BindToInput<IDurableEntityClient>(this.GetClient);
             rule.BindToInput<IDurableClient>(this.GetClient);
 
+            // We add a binding rule to support the now deprecated `orchestrationClient` binding
+            // A cleaner solution would be to have the prior rule have an OR-style selector between
+            // DurableClientAttribute and OrchestrationClientAttribute, but it's unclear if that's
+            // possible (for now).
+#pragma warning disable CS0618 // Type or member is obsolete
+            var backwardsCompRule = context.AddBindingRule<OrchestrationClientAttribute>()
+#pragma warning restore CS0618 // Type or member is obsolete
+                .AddConverter<string, StartOrchestrationArgs>(bindings.StringToStartOrchestrationArgs)
+                .AddConverter<JObject, StartOrchestrationArgs>(bindings.JObjectToStartOrchestrationArgs)
+                .AddConverter<IDurableClient, string>(bindings.DurableOrchestrationClientToString);
+
+            backwardsCompRule.BindToCollector<StartOrchestrationArgs>(bindings.CreateAsyncCollector);
+            backwardsCompRule.BindToInput<IDurableOrchestrationClient>(this.GetClient);
+            backwardsCompRule.BindToInput<IDurableEntityClient>(this.GetClient);

```

Comment:
Note that if you go the route where you get rid of `backwardsCompRule`, you can disregard this feedback.

Reply:
We already discourage users from using `[OrchestrationClient]` by putting the `[Obsolete]` attribute on it.  But the whole reason for `backCompRule` is that inheritance by itself is not sufficient, per David's investigation last week.

Reply:
Good to know inheritance didn't work. I must either be misremembering the behavior of binding rules, or they have changed since I last looked at the code.

In that case, while I still don't see a good reason to include support for the backwards compatible binding with a V2 only feature, it probably doesn't hurt to include the support, so this nit can be disregarded.